lib/chef.rb

```diff
@@ -17,6 +17,9 @@
 #
 
 require "chef/version"
+
+require "chef_core/text"
```

Comment:
Do we need this require at this top level when we require it in knife bootstrap as well?

Reply:
We cdan remove it from knife bootstrap. 

This is a complicated one.  Text initializes r18n, which in turn uses the YAML lib.
However, if we hit a path that has us including rubygems first, it monkey-patches YAML 
constants (Psych) for backwards compatibility in a way that makes r18n raise an error. 

This makes it so that the  only way to actually load chef/text is  to ensure we do so before rubygems is used to parse a gemspec.

Reply:
probably shouldn't remove it from knife bootstrap.  generally we have to assume that files can get required by client code without this top level require having been hit, so we'd need it both places.

Reply:
(there should probably be a comment here about why this line is here)